source/Calamari.Tests/Fixtures/Conventions/JsonConfigurationVariablesConventionFixture.cs

```diff
@@ -16,87 +14,40 @@ namespace Calamari.Tests.Fixtures.Conventions
     public class JsonConfigurationVariablesConventionFixture
     {
         RunningDeployment deployment;
-        IStructuredConfigVariableReplacer configVariableReplacer;
-        ICalamariFileSystem fileSystem;
-        ILog log;
-        const string StagingDirectory = "C:\\applications\\Acme\\1.0.0";
+        IStructuredConfigVariablesService service;
 
         [SetUp]
         public void SetUp()
         {
             var variables = new CalamariVariables(); 
-            variables.Set(KnownVariables.OriginalPackageDirectoryPath, TestEnvironment.ConstructRootedPath("applications", "Acme", "1.0.0"));
+            service = Substitute.For<IStructuredConfigVariablesService>();
             deployment = new RunningDeployment(TestEnvironment.ConstructRootedPath("Packages"), variables);
-            configVariableReplacer = Substitute.For<IStructuredConfigVariableReplacer>();
-            fileSystem = Substitute.For<ICalamariFileSystem>();
-            fileSystem.DirectoryExists(Arg.Any<string>()).Returns(false);
-            log = Substitute.For<ILog>();
         }
 
         [Test]
         public void ShouldNotRunIfVariableNotSet()
         {
-            var service = new StructuredConfigVariablesService(configVariableReplacer, fileSystem, log);
             var convention = new JsonConfigurationVariablesConvention(service);
             convention.Install(deployment);
-            configVariableReplacer.DidNotReceiveWithAnyArgs().ModifyFile(null, null);
+            service.DidNotReceiveWithAnyArgs().ReplaceVariables(deployment);
         }
 
         [Test]
-        public void ShouldFindAndCallModifyOnTargetFile()
```

Comment:
This case is covered pretty well in `DeployPackageWithStructuredConfigurationFixture`